examples/Getting Started with Semiconductor Module/Python/Getting Started with Semiconductor Module.py

```diff
@@ -0,0 +1,270 @@
+    pins,
+):
+    """Measure continuity on configured Pins (niDcPower, niDigital)."""
+    dcpower_filtered_pins = tsm_context.filter_pins_by_instrument_type(
+        pins, InstrumentTypeIdConstants.NI_DCPOWER, Capability.ALL
+    )
+    pin_query, sessions, channel_strings = tsm_context.pins_to_nidcpower_sessions(
+        dcpower_filtered_pins
+    )
+    per_dcpower_session_measurements = []
+    dcpower_continuity_measurements = []
+    for session, channel_string in zip(sessions, channel_strings):
+        assert isinstance(session, nidcpower.Session)
+        assert isinstance(channel_string, str)
+        # call some methods on the session to ensure no errors
+        pin_session = session.channels[channel_string]
+        session.abort()
+        pin_session.output_function = nidcpower.OutputFunction.DC_CURRENT
+        pin_session.current_level = 10e-3
+        pin_session.output_enabled = True
+        pin_session.source_delay = 250e-6
+        session.initiate()
+        session.wait_for_event(nidcpower.Event.SOURCE_COMPLETE)
+        per_dcpower_session_measurements = pin_session.measure_multiple()
+
+        # simulate continuity data for each channel
+        per_dcpower_session_measurements = []
+        for _ in channel_string.split(","):
+            per_dcpower_session_measurements += (
+                (((0.75 - -0.75) * gauss(0, 0.17)) + (((0.75 - -0.75) / 2) + -0.75)),
+            )
+
+        dcpower_continuity_measurements.append(per_dcpower_session_measurements)
+
+    pin_query.publish(dcpower_continuity_measurements)
+
+    digital_filtered_pins = tsm_context.filter_pins_by_instrument_type(
+        pins, InstrumentTypeIdConstants.NI_DIGITAL_PATTERN, Capability.ALL
+    )
+    pin_query, sessions, pin_set_strings = tsm_context.pins_to_nidigital_sessions_for_ppmu(
+        digital_filtered_pins
+    )
+    per_digital_session_measurements = []
+    digital_continuity_measurements = []
+    for session, pin_set_string in zip(sessions, pin_set_strings):
+        assert isinstance(session, nidigital.Session)
+        assert isinstance(pin_set_string, str)
+
+        # call some methods on the session to ensure no errors
+        session.ppmu_output_function = nidigital.PPMUOutputFunction.VOLTAGE
+        session.ppmu_voltage_level = 10e-3
+        session.ppmu_source()
+        per_digital_session_measurements = session.ppmu_measure(
+            nidigital.PPMUMeasurementType.VOLTAGE
+        )
+        session.abort
+
+        # simulate continuity data
+        per_digital_session_measurements = []
+        for _ in pin_set_string.split(","):
+            per_digital_session_measurements += (
+                (((0.75 - -0.75) * gauss(0, 0.17)) + (((0.75 - -0.75) / 2) + -0.75)),
+            )
+
+        digital_continuity_measurements.append(per_digital_session_measurements)
+
+    pin_query.publish(digital_continuity_measurements)
+
+
+@nitsm.codemoduleapi.code_module
+def leakage(
```

Comment:
oh snap. we don't have the proper get specifications methods to fully realize this function. that's not good. we should probably add those before merging this PR.  I'll get started on that while some of the other review comments are addressed.